src/index.ts

```diff
@@ -58,14 +58,20 @@ const createPlugin = (options: ServiceWorkerConfig = {}): AstroIntegration => {
   return {
     name: PKG_NAME,
     hooks: {
-      "astro:config:setup": ({ command, injectRoute, injectScript }) => {
+      "astro:config:setup": ({
+        command,
+        injectRoute,
+        injectScript,
+        config,
+      }) => {
         const autoRegister = options.registration?.autoRegister ?? true;
         if (autoRegister) {
+          const swPath = join(config.base, SW_NAME);
           injectScript(
             "head-inline",
             `\
 if ('serviceWorker' in navigator) {
-  navigator.serviceWorker.register('/${SW_NAME}');
+  navigator.serviceWorker.register('${swPath}');
```

Comment:
The two usages below need to be updated too, line 82 and 105

Reply:
It has nothing to do with the file path,
The base path will need to change when the site is not deployed on root.

for example,

I build the project with base path `/test` for Github Pages, it will be

```tree
/dist
- index.html
- service-worker.js
```

but the path is not deployed to `https://ntsd.github.io` but `https://ntsd.github.io/test` (`test` is the repo name)

If we register `/service-worker.js` it will response 404 because it's looking for `/service-worker.js (https://ntsd.github.io/service-worker.js)`  not `/test/service-worker.js`

more details: <https://docs.astro.build/en/guides/deploy/github/#how-to-deploy>

Reply:
The path on line 82 and the service worker register path need to be in sync